// fullsession-heatmap-package/dist/heatmapDrawer.js
import { h337 } from "heatmap.js";
export class heatmapDrawer {
    constructor(clickHeatMap, movementHeatMap, htmlElement) {
        var box = htmlElement.getBoundingClientRect();
        var hm = document.createElement('div');
        document.body.appendChild(hm);
        hm.style.top = "0";
        hm.style.left = "0";
        hm.style.width = box.width + 'px';
        hm.style.height = Math.min(5000, box.height) + 'px';
        hm.style.zIndex = String(Number.MAX_SAFE_INTEGER);
        this.heatmap = h337.create({
            container: hm,
            radius: 60,
            maxOpacity: .4,
            blur: .70,
        });
        document.querySelector('html').style.position = 'relative';
        hm.style.position = 'absolute';
        this.drawHeatMap(clickHeatMap, movementHeatMap);
    }
    drawHeatMap(clickHeatMap, movementHeatMap) {
        const start = Date.now();
        this.heatmap.addData(movementHeatMap);
        this.heatmap.addData(clickHeatMap);
        const duration = Date.now() - start;
    }
}
